$NetBSD: patch-aa,v 1.10 2007/11/04 15:34:09 rillig Exp $

* "grep -q" doesn't work everywhere.
* Do not use the home directory of the user building the package.

--- util/fluxbox-generate_menu.in.orig	2007-10-08 18:49:08.000000000 +0000
+++ util/fluxbox-generate_menu.in	2007-11-04 15:31:20.000000000 +0000
@@ -51,7 +51,7 @@ WHOAMI=`whoami`
 [ "$WHOAMI" = root ] && PATH=/bin:/usr/bin:/usr/local/bin:/usr/X11R6/bin
 
 # Check for Imlib2-support
-if fluxbox -info 2> /dev/null | grep -q "^IMLIB"; then
+if fluxbox -info 2> /dev/null | grep "^IMLIB" > /dev/null; then
     PNG_ICONS="yes"
 else
     # better assume to assume "no"
@@ -338,7 +338,7 @@ searchForIcon(){
     # echo "^.${entry_exec}.[[:space:]]*<.*/${icon_base}\....>" 
     if [ -f "$entry_icon" ]; then
     # if icon exists and entry does not already exists, add it
-        if ! grep -q -m 1 "^.${entry_exec}.[[:space:]]*<.*/${icon_base}\....>" $ICONMAPPING 2> /dev/null; then
+        if ! grep "^.${entry_exec}.[[:space:]]*<.*/${icon_base}\....>" $ICONMAPPING > /dev/null; then
             echo -e "\"${entry_exec}\" \t <${entry_icon}>" >> $ICONMAPPING
         else 
             : echo "#    mapping already exists for ${entry_exec}" >> $ICONMAPPING
@@ -1210,7 +1210,7 @@ case ${LC_ALL} in
 esac
 
 # Set Defaults
-USERFLUXDIR="${HOME}/.@pkgprefix@fluxbox@pkgsuffix@"
+USERFLUXDIR="\${HOME}/.@pkgprefix@fluxbox@pkgsuffix@"
 MENUFILENAME="${MENUFILENAME:=${USERFLUXDIR}/menu}"
 MENUTITLE="${MENUTITLE:=Fluxbox}"
 HOMEPAGE="${HOMEPAGE:=fluxbox.org}"
@@ -1708,7 +1708,7 @@ append_submenu "${MULTIMEDIAMENU}"
        append_submenu "${XUTILSMENU}"
            normal_find xfontsel xman xload xbiff editres viewres xclock \
                xmag wmagnify gkrellm gkrellm2 vmware portagemaster agave 
-           find_it xrdb append "[exec] (Reload .Xdefaults) {xrdb -load $HOME/.Xdefaults}"
+           find_it xrdb append "[exec] (Reload .Xdefaults) {xrdb -load \$HOME/.Xdefaults}"
        append_menu_end
 append_menu_end
 
